p2p/protocol/identify/id.go

```diff
@@ -193,7 +216,7 @@ func (ids *IDService) IdentifyConn(c network.Conn) {
 	s.SetProtocol(ID)
 
 	// ok give the response to our handler.
-	if err := msmux.SelectProtoOrFail(ID, s); err != nil {
+	if err = msmux.SelectProtoOrFail(ID, s); err != nil {
```

Comment:
hm. I'm usually a `if err :=` or break it into two lines kind of guy. Side-effects from an if statement make me unhappy.